src/CVNBot/ListManager.cs

```diff
@@ -144,10 +155,15 @@ public string AddUserToList(string name, string project, UserType type, string a
                 // Original type was same as new type; update list with new details
                 using (IDbCommand cmd = dbcon.CreateCommand())
                 {
-                    cmd.CommandText = "UPDATE users SET adder = '" + adder.Replace("'", "''") + "', reason = '"
-                        + reason.Replace("'", "''") + "', expiry = '" + GetExpiryDate(expiry)
-                        + "' WHERE name = '" + name.Replace("'", "''") + "' AND project = '" + project
-                        + "' AND type ='" + ((int)originalType).ToString() + "'";
+                    cmd.CommandText = "UPDATE users SET adder = '@adder', reason = '@reason', expiry = '@expiry' WHERE name = '@name' AND project = '@project' AND type ='@originalType'";
+                    cmd.Parameters.Add(new SqliteParameter("@adder", adder.Replace("'", "''")));
```

Comment:
Are these replacements still needed? I would expect them to potentially (wrongly) lead to real modification of input. Or was that already behaving incorrectly before? (I assumed they were escapes).

Reply:
I'm not sure about the replacements, I just moved them over from the original code figuring they were being stored in the DB that way and thus the formatting was necessary when looking up or necessary.

Reply:
You are right, those replacements are not needed and neither are the single quotes in the commandText. You can reject this PR and Ill send in another one with the updates.

Reply:
I understand that it was a real modification on the string, but there were no prepared or bound values previously. The modified string became a direct part of the SQL command.

As such it don't think it was modififying the value in a way that would be seen by the database, rather it was escaping it so that it would look identical to the database.

If you send `SET foo = 'don''t try this at home'` the value will be `don't try this at home`.

Such manual escaping is a bad practice, so I like your change. But keeping the replaces meant is will actually start changing the values.

If you prefer a new PR that's okay, but I would also be fine with updating this one. Either push a fixup or amend it, I'll squash-merge either way :)

Thanks!

Reply:
No worries about, sorry for the confusion :)....Had to look up the docs as well.

Just did an amend.